src/index.ts

```diff
@@ -9,15 +9,15 @@ export class FunctionsClient {
   constructor(
     url: string,
     {
-      headers = {},
+      headers,
       customFetch,
     }: {
       headers?: Record<string, string>
       customFetch?: Fetch
-    }
+    } = {}
   ) {
     this.url = url
-    this.headers = headers
+    this.headers = headers ?? {}
```

Comment:
```suggestion
    this.headers = headers
```
for consistency I'd leave this